readers/combined.hh

```diff
@@ -49,15 +49,18 @@ mutation_reader make_combined_reader(schema_ptr schema,
         reader_permit permit,
         std::vector<mutation_reader>,
         streamed_mutation::forwarding fwd_sm = streamed_mutation::forwarding::no,
-        mutation_reader::forwarding fwd_mr = mutation_reader::forwarding::yes);
+        mutation_reader::forwarding fwd_mr = mutation_reader::forwarding::yes,
+        combined_reader_statistics* statistics = nullptr);
```

Comment:
Wouldn't this be the right spot to allocate the vector? Or at least resize it? To avoid a mismatch.

Reply:
I can definitely resize it here but that is not enough as `sstable_set::make_local_shard_sstable_reader` calls the variant with `std::unique_ptr<reader_selector>`.

So instead, the plan is to extend `reader_selector` with the information on the maximum readers count and use it to resize the statistics vector. This way, it will be done exactly in one place that actually creates the reader.

Reply:
Ok. Note that having multiple resizes isn't a problem, as long as it's not in the fast path.